src/parser.rs

```diff
@@ -124,3 +126,136 @@ impl Statement {
         }
     }
 }
+
+#[cfg(test)]
+mod statement_prep {
+    use super::*;
+
+    macro_rules! get_statement {
+        ($input: literal) => {
+            Statement::prep(&$input.to_owned());
+        };
+    }
+
+    #[test]
+    fn test_returns_unk_on_unknown_statement() {
+        let statement = get_statement!("UNKNOWN_STATEMENT");
+        assert_eq!(statement, Statement {
+            stype: StatementType::Unk,
+            key: None,
+            value: None
+        });
+    }
+
+    mod get {
+        use super::*;
+
+        #[test]
+        fn returns_fail_on_get_without_key() {
+            let statement = get_statement!("GET");
+            assert_eq!(statement, Statement {
+                stype: StatementType::Fail,
+                key: None,
+                value: None
+            });
+        }
+
+        #[test]
+        fn returns_get() {
```

Comment:
the same here, renaming `returns_get()` as `test_parsing_proper_get_statement()` or something similar, provides better description of what the test does according to me...